src/content/symbol-store-db.js

```diff
@@ -1,134 +1,203 @@
+        req.onsuccess = () => {
+          const cursor = (req.result: any);
+          if (cursor) {
+            const value = (cursor: IDBCursorWithValue).value;
+            value.lastUsedDate = new Date();
+            const updateDateReq = store.put(value, cursor.primaryKey);
+            const { addrs, index, buffer } = value;
+            updateDateReq.onsuccess = () => resolve([addrs, index, buffer]);
+          } else {
+            reject(new Error('The requested library does not exist in the database.'));
+          }
+        };
+      });
     });
   }
 
-  importLibrary(debugName, breakpadId, [addrs, index, buffer]) {
-    if (!this._db) {
-      return this._setupDBPromise.then(() => this.importLibrary(debugName, breakpadId, [addrs, index, buffer]));
-    }
-
+  _setupDB(dbName: string): Promise<IDBDatabase> {
     return new Promise((resolve, reject) => {
-      const transaction = this._db.transaction('symbol-tables', 'readwrite');
-      transaction.onerror = reject;
-      const tableStore = transaction.objectStore('symbol-tables');
-      const putReq = tableStore.put({ debugName, breakpadId, addrs, index, buffer });
-      putReq.onsuccess = () => {
-        resolve(putReq.result);
+      const openReq = window.indexedDB.open(dbName, 5);
+      openReq.onerror = () => reject(openReq.error);
+      openReq.onupgradeneeded = () => {
+        const db = openReq.result;
+        db.onerror = reject;
+        db.deleteObjectStore('symbol-tables');
+        const store = db.createObjectStore('symbol-tables', { autoIncrement: true });
+        store.createIndex('libKey', ['debugName', 'breakpadId'], { unique: true });
+        store.createIndex('lastUsedDate', 'lastUsedDate', { unique: false });
+      };
+      openReq.onsuccess = () => {
+        const db = openReq.result;
+        resolve(db);
+        this._deleteAllBeforeDate(db, new Date(+new Date - this._maxAge));
       };
     });
   }
 
-  getFuncAddressTableForLib(libKey) {
-    if (!this._db) {
-      return this._setupDBPromise.then(() => this.getFuncAddressTableForLib(libKey));
+  // Many of the utility functions below use callback functions instead of
+  // promises. That's because IndexedDB transactions auto-close at the end of
+  // the current event tick if there hasn't been a new request after the last
+  // success event. So we need to synchronously add more work inside the
+  // onsuccess handler, and we do that by calling the callback function.
+  // Resolving a promise only calls any then() callback at the next microtask,
+  // and by that time the transaction will already have closed.
+  // We don't propagate errors because those will be caught by the onerror
+  // handler of the transaction that we got `store` from.
+  _deleteRecords(store: IDBObjectStore, primaryKeys: number[], callback: () => void): void {
+    const leftToDelete = primaryKeys.slice(0);
+    function deleteNext() {
+      if (leftToDelete.length === 0) {
+        callback();
```

Comment:
nit: use something like `Promise.resolve().then(callback)` so that the callback is consistently called in an asynchronous way, even when there is nothing to delete.